Code/Framework/AtomCore/AtomCore/Instance/InstanceDatabase.h

```diff
@@ -368,7 +337,19 @@ namespace AZ
         template<typename Type>
         Data::Instance<Type> InstanceDatabase<Type>::Create(const Asset<AssetData>& asset, const AZStd::any* param)
         {
-            return FindOrCreate(Data::InstanceId::CreateRandom(), asset, param);
+            const InstanceId id = InstanceId::CreateRandom();
```

Comment:
You could move the ID creation to the end of the function so that it's only done if the asset successfully loads